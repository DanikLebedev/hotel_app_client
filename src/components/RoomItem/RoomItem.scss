@import 'src/index';

.rooms-page__item {
 box-shadow: 5px 5px 5px 5px #cdcdcd;
  padding: 30px;
  transition: .3s ease-in-out;

  p {
    margin-right: 10px;
  }

  &:hover {
    color: #ffffff;
    background: $price-color;
  }
}

.room-page__item-img {
  border-radius: 20px;
}

.rooms-page__wrapper{
  &:nth-child(odd) {
    animation: slideInRight .7s ease-in-out;
  }
  &:nth-child(even) {
    animation: slideInLeft .7s ease-in-out;
  }
}


@keyframes slideInLeft {
  from {
    -webkit-transform: translate3d(-100%, 0, 0);
    transform: translate3d(-100%, 0, 0);
    visibility: visible;
  }

  to {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes slideInRight {
  from {
    -webkit-transform: translate3d(100%, 0, 0);
    transform: translate3d(100%, 0, 0);
    visibility: visible;
  }

  to {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}


@keyframes slideIn {
  from {
    transform: translateY(30px);
    opacity: 0;
  }
  to {
    transform: translate(0);
    opacity: 1;
  }

}

@media screen and (min-width: 768px) and (max-width: 1199px) {
  .rooms-page__item {
    &-text {
      display: flex;
      flex-direction: column;
    }
  }
}

