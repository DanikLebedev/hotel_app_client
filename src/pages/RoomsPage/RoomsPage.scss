@import "src/index";

.room-page {
  &-bg {
    background: url("../../assets/images/dui21-deluxe-room1.add1.low-res.jpg") center center / cover;
    height: 50vh;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -3;
  }
  h1 {
    color: $base-text-color;
    font-size: 4em;
  }
  &-wrapper {
    padding-top: 35vh;
  }
}

@media screen
and (min-width: 768px)
and (max-width: 1199px){
  .room-page {
    h1 {
      font-size: 3em;
    }
   &-wrapper {
     padding-top: 40vh;
   }
  }
}


@media screen and (min-width: 300px) and (max-width: 480px) {
  .room-page {
    h1 {
      font-size: 2em;
    }
    &-wrapper {
      padding-top: 20vh;
    }
  }
}
