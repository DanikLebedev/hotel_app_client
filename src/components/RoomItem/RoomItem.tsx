import React, {PropsWithChildren} from 'react'
import {config} from '../../config'
import {Room} from "../../interfaces/clientInterfaces";
import {Col, Container, Row} from "react-bootstrap";
import './RoomItem.scss'



export const RoomItem = (props: {roomInfo: Room}) => {
    return (
       <Container>
           <Row className='rooms-page__item mb-3'>
               <Col lg={6}>
                   <img style={{width: '80%', height: '80%'}} src={config.baseUrl + props.roomInfo.image } alt="img"/>
               </Col>
               <Col lg={6}>
                   <h3>{props.roomInfo.title}</h3>
                   <div className='d-flex'>
                       <p>Category: {props.roomInfo.category}</p>
                       <p>Price: {props.roomInfo.price}</p>
                   </div>
                    <div  className='d-flex'>
                        <p>Area: {props.roomInfo.area}</p>
                        <p>Guests: {props.roomInfo.guests}</p>
                        <p>Rooms: {props.roomInfo.rooms}</p>
                    </div>
                   <p>Description: {props.roomInfo.description}</p>
                   <button className='button btn-black'>Book Room</button>
               </Col>
           </Row>
       </Container>
    )
}

export default RoomItem