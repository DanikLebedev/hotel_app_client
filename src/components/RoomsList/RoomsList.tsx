import React, {useCallback, useEffect, useState} from 'react'
import RoomItem from "../RoomItem/RoomItem";
import Loader from "../Loader/Loader";
import {RoomService} from "../../APIServices/roomService";
import {Room} from "../../interfaces/clientInterfaces";

export const RoomsList: React.FC = () => {
    const [fetchedRooms, setFetchedRooms] = useState<Room[]>([])

    const fetchRoom = useCallback( () => {
        RoomService.getAllRooms().then(({rooms}) => setFetchedRooms(rooms))
    },[])

    useEffect(() => {
        fetchRoom()
    }, [fetchRoom])

    return (
        <div>
            {!fetchedRooms[0] ? <Loader/> : fetchedRooms.map((room: Room, i: number) => {
                return <RoomItem key={room.title + i} roomInfo={room}/>
            })}
        </div>
    )
}

export default RoomsList