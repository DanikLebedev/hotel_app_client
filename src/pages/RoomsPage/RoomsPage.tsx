import React from 'react'
import RoomsList from '../../components/RoomsList/RoomsList'
import {Container} from "react-bootstrap";
import './RoomsPage.scss'

export const RoomsPage = () => {
    return (
        <div className='room-page'>
            <div className='room-page-bg'></div>
            <Container className='room-page-wrapper'>
                <h1>Book Room</h1>
                <RoomsList/>
            </Container>

        </div>
    )
}

export default RoomsPage