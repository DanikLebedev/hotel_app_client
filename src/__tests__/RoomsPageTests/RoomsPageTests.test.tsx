import { shallow } from 'enzyme';
import React from 'react';
import RoomsPage from '../../pages/RoomsPage/RoomsPage';
import RoomsList from '../../components/RoomsList/RoomsList';
import RoomItem from '../../components/RoomItem/RoomItem';

const roomData = {
    area: 0,
    category: '',
    description: '',
    guests: 0,
    image: '',
    isBooked: false,
    price: 0,
    rooms: 0,
    title: '',
    _id: '',
};

describe('rooms page test tests', () => {
    it('should renders room page without crashing', function() {
        const wrapper = shallow(<RoomsPage />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should renders rooms list without crashing', function() {
        const wrapper = shallow(<RoomsList />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should renders rooms item without crashing', function() {
        const wrapper = shallow(<RoomItem roomInfo={roomData} searchRoom={false} />);
        expect(wrapper).toMatchSnapshot()
    });
});
