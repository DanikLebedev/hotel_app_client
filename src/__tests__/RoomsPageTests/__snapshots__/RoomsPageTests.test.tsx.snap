// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rooms page test tests should renders room page without crashing 1`] = `
<div
  className="room-page"
>
  <withI18nextTranslation(FindRoomForm) />
  <div
    className="room-page-bg d-flex justify-content-center align-items-end"
  >
    <h1>
      Book Room
    </h1>
  </div>
  <Container
    className="room-page-wrapper"
    fluid={false}
  >
    <t
      afterLoad={[Function]}
      beforeLoad={[Function]}
      useIntersectionObserver={true}
      visibleByDefault={false}
    >
      <RoomsList />
    </t>
  </Container>
</div>
`;

exports[`rooms page test tests should renders rooms item without crashing 1`] = `
<Container
  className="rooms-page__wrapper"
  fluid={false}
>
  <ForwardRef
    className="rooms-page__item mb-3"
    noGutters={false}
  >
    <Col
      className="room-page__item-img"
      lg={6}
      md={6}
      sm={6}
      style={
        Object {
          "background": "url(http://localhost:5000/static/) center center / cover",
        }
      }
      xs={6}
    />
    <Col
      lg={6}
      md={6}
      sm={6}
      xs={6}
    >
      <h3 />
      <div
        className="rooms-page__item-text"
      >
        <p>
          Category: 
        </p>
        <p>
          Price: 
          0
          $
        </p>
      </div>
      <NavLink
        className="button btn-black"
        id="redirect-button"
        to="/rooms/"
      >
        Show More
      </NavLink>
    </Col>
  </ForwardRef>
</Container>
`;

exports[`rooms page test tests should renders rooms list without crashing 1`] = `
<div>
  <RoomItem
    key="0"
    roomInfo={
      Object {
        "_id": "",
        "area": 0,
        "category": "",
        "description": "",
        "guests": 0,
        "image": "",
        "isBooked": false,
        "price": 0,
        "rooms": 0,
        "title": "",
      }
    }
    searchRoom={false}
  />
</div>
`;
